docs/modify_fields.md

```diff
@@ -0,0 +1,123 @@
+# `modify_fields` Processor
+
+The `modify_fields` processor allows customization of the structure and contents of log entries.
+
+## Configuration structure
+
```

Comment:
[Optional] Add a sentence to briefly explain the "source" section and "mutation" section?

It is clear to me, but I have a lot of context for this functionality already. Figured it might be helpful to someone who is new to the `modify_fields` processor.